.gitlab/functional_test/regression_detector.yml

```diff
@@ -17,21 +17,15 @@ single-machine-performance-regression_detector:
       - outputs/report.md # for debugging, also on S3
       - outputs/regression_signal.json # for debugging, also on S3
       - outputs/bounds_check_signal.json # for debugging, also on S3
+      - outputs/quality_gates_report.json # for debugging, also on S3
       - outputs/junit.xml # for debugging, also on S3
+      - outputs/report.json # for debugging, also on S3
+      - outputs/decision_record.md # for posterity, this is appended to final PR comment
     when: always
   variables:
     SMP_VERSION: 0.18.1
-  # At present we require two artifacts to exist for the 'baseline' and the
-  # 'comparison'. We are guaranteed by the structure of the pipeline that
-  # 'comparison' exists, not so much with 'baseline' as it has to come from main
-  # merge pipeline run. This is solved in datadog-agent by updating a file in S3
-  # with the SHA of the merge base from main. It's solved in Vector by
-  # building Vector twice for each Regression Detector run.
-  #
-  # We allow failure for now. _Unfortunately_ this also means that if the
-  # Regression Detector finds a performance issue with a PR it will not be
-  # flagged.
-  allow_failure: true
+  # See 'decision_record.md' for the determination of whether this job passes or fails.
+  allow_failure: false
```

Comment:
In case someone else reviews this PR: the default value of `allow_failure` actually changes, depending on whether a job is run manually (default is `true`) or not (default is `false`). So I support explicitly setting `allow_failure` here.